app/src/main/java/in/aviaryan/hinix/GameBoard.java

```diff
@@ -39,6 +39,78 @@ public GameBoard(InputStream wordListStream) throws IOException {
         }
     }
 
+    private void clearBoard() {
+        for (int i = 0; i < chars.length; i++) {
+            for (int j = 0; j < chars[i].length; j++) {
+                chars[i][j] = '\0';
+            }
+        }
+    }
+
+    private boolean findPlaceForWord(String word, HashSet<String> addedWords,
+                                     ArrayList<String> addedWordsList) {
+        int shortRun = 0;
+        int i = 0, j = 0;
+        for (; i < chars.length; i++) {
+            for (; j < chars[i].length; j++){
+                if (shortRun == 0) {
+                    int posArr[] = findSameCharGrid(word.charAt(0));
+                    if (posArr[0] != -1) {
+                        i = posArr[0]; j = posArr[1]; shortRun = 1;
+                    } else {
+                        shortRun = 2;
+                    }
+                }
+                if (chars[i][j] == '\0' || chars[i][j] == word.charAt(0)){
+                    HashSet<String> visited = new HashSet<>();
+                    ArrayList<String> order = new ArrayList<>();
+                    order.add(i + " " + j);
+                    visited.add(i + " " + j);
+                    boolean success = fitWord(word.substring(1), i, j, visited, order);
+                    if (success){
+                        wordPlaceTries = 0;
+                        int count = 0;
+                        addedWords.add(word);
+                        addedWordsList.add(word);
+                        // add word to board
+                        if (solutionList.size() != word.length()){
+                            Log.v(LOG_TAG, word + " " + solutionList.toString());
+                        }
+                        for (String s: solutionList){
+                            int x = Integer.parseInt(s.split("\\s+")[0]);
+                            int y = Integer.parseInt(s.split("\\s+")[1]);
+                            chars[x][y] = word.charAt(count);
+                            count++;
+                        }
+                        return true;
+                    }
+                }
+                if (shortRun == 1){
+                    shortRun = 2; i=0; j=-1;
+                }
+            }
+        }
+        return false;
+    }
+
+    private boolean hasBadLetters(String word, String badLetters) {
```

Comment:
Can you please move badLetters inside this function and not pass it as a parameter. Or better, make it a final class variable.